test/helpers.py

```diff
@@ -0,0 +1,75 @@
+import pytest
+from httpx import AsyncClient
+
+
+def generate_authorization_header(access_token: str) -> dict:
+    return {"Authorization": f"Bearer {access_token}"}
+
+
+async def get_access_token(client: AsyncClient, email: str = "email@example.com") -> str:
+    await client.post(
+        "/users",
+        json={
+            "email": email,
+            "password": "String123",
+            "full_name": "Minh Hoang",
+        },
+    )
+    response = await client.post(
+        "/users/auth",
+        json={
+            "email": email,
+            "password": "String123",
+        },
+    )
+    return response.json().get("access_token")
+
+
+@pytest.fixture
+async def access_token(client: AsyncClient) -> str:
+    return await get_access_token(client)
+
+
+@pytest.fixture
+async def access_token_other_user(client: AsyncClient) -> str:
+    return await get_access_token(client, "other@email.com")
+
+
+@pytest.fixture
+async def create_category(client: AsyncClient, access_token: str) -> None:
+    await client.post(
+        "/categories",
+        headers={"Authorization": f"Bearer {access_token}"},
+        json={"name": "Car", "description": "Car has 4 wheels"},
+    )
+
+
+@pytest.fixture
+async def create_item(
+    client: AsyncClient,
+    access_token: str,
+    create_category: None,
+) -> None:
+    await client.post(
+        "/categories/1/items",
+        headers={"Authorization": f"Bearer {access_token}"},
+        json={
+            "name": "Volvo",
+            "description": "Volvo from Germany",
+        },
+    )
+
+
+@pytest.fixture
+async def create_many_items(
+    client: AsyncClient,
+    access_token: str,
+    create_category: None,
+):
+    # Create 30 items to add to category
+    for time in range(30):
+        await client.post(
+            "/categories/1/items",
+            headers={"Authorization": f"Bearer {access_token}"},
+            json={"name": "Volvo" + str(time), "description": "Volvo from Germany"},
+        )
```

Comment:
Do these functions need to run before every test case. If not, please remove `@pytest.fixture`. I think this can explain why we have a very poor performance when running tests.

Reply:
These fixtures are at scope of function by default and we don't specify `autouse` on them, so they only run when requested by a test (ref: https://docs.pytest.org/en/6.2.x/fixture.html#requesting-fixtures).

The only fixture that using `autouse=True` is the one that re-initialize tables in database. When I remove `autouse`, the testing time reduces from 45s to 20s.